test/models/weather_request_test.rb

```diff
@@ -34,4 +36,40 @@ def setup
       @weather_request.destroy
     end
   end
+  
+  test "should extract weather info" do
+    expect_info = OpenStruct.new(weather_time: 'Wed, 24 Oct 2018 06:00 PM AEDT', temperature: 17, feels_like: 16, description: 'Cloudy', sunrise: '6:4 am', sunset: '7:17 pm')
+    Net::HTTP.expects(:get).returns(@json_valid)
+    weather_request = WeatherRequestService.new(@location).run
+    weather_info = weather_request.weather_info
+    assert_equal weather_info.weather_time, expect_info.weather_time
+    assert_equal weather_info.temperature,  expect_info.temperature
+    assert_equal weather_info.feels_like,   expect_info.feels_like
+    assert_equal weather_info.description,  expect_info.description
+    assert_equal weather_info.sunrise,      expect_info.sunrise
+    assert_equal weather_info.sunset,       expect_info.sunset
```

Comment:
What would be the advantage or disdavantage of doing the following?
```ruby
    assert_equal [
      expect_info.weather_time,
      expect_info.temperature,
      expect_info.feels_like,
      expect_info.description,
      expect_info.sunrise,
      expect_info.sunset
    ],
    [
      weather_info.weather_time,
      weather_info.temperature,
      weather_info.feels_like,
      weather_info.description,
      weather_info.sunrise,
      weather_info.sunset
    ]
```

Reply:
I'd say your example would probably run faster.  It would also technically mean less assertions per test, which some people care about from a style point of view.  I think it might be a disadvantage if the model had 20+ attributes, but it doesn't :)  Can't think of anything else.

Reply:
The main difference I'm thinking of here is that you don't get to see the remaining assertions' results.  So, for example, if `weather_time` failed the assertion, you don't get to see any of the subsequent values.  If you saw all the attributes in one go, (rather than trying to fix the weather time one first), then you might get a better picture of why any or all might be failing.

Note, Minitest assertions are designed so that the first parameter is the expected value; "backwards" compared to RSpec.

Reply:
Cool.  thanks.